src/pages/Authentication/Login.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useEffect } from "react";
+import { NavLink, useLocation, useNavigate } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { authenticationLoginThunk } from "reduxFiles";
+import styles from "./authentication.module.css";
+import { initialLoginData, guestLoginCredentials } from "constants";
+import { useAlerts } from "hooks";
+
+const Login = () => {
+  const { isUserLoggedIn } = useSelector(
+    (storeReceived) => storeReceived.authenticationStore
+  );
+  const { showAlerts } = useAlerts();
+  const [loginCredentials, setLoginCredentials] = useState(initialLoginData);
+  const { email: emailInput, password: passwordInput } = loginCredentials;
+  const dispatch = useDispatch();
+  const setDataFromInput = (eventReceived) => {
+    setLoginCredentials((prev) => ({
+      ...prev,
+      [eventReceived.target.name]: eventReceived.target.value,
+    }));
+  };
+  const location = useLocation();
+  const navigate = useNavigate();
+  let from = location.state?.from?.pathname ?? "/explore";
+  const loginFormSubmissionHandler = async (userDetailsReceived) => {
+    try {
+      const submissionResponse = await dispatch(
+        authenticationLoginThunk(userDetailsReceived)
+      );
+      const { encodedToken: gatheredEncodedToken } =
+        submissionResponse?.payload;
+      if (submissionResponse?.error) {
+        throw new Error(submissionResponse?.error);
+      }
+      if (gatheredEncodedToken) {
+        showAlerts("success", "Logged in successfully");
+        navigate(from, { replace: true });
+      }
+    } catch (submissionError) {
+      showAlerts("error",`Login failed`);
+    }
+  };
+
+  useEffect(() => {
+    if (isUserLoggedIn) {
+      navigate(-1);
```

Comment:
what does `navigate(-1)` do?

Reply:
Suppose if a logged in user tries to access the login and signup routes then he would be redirected to the route on which he was there before going to the login/signup routes. That why -1 is there to navigate the user one step behind instead of going to login/signup